warehouse/query-core/src/main/java/datawave/query/attributes/Content.java

```diff
@@ -157,12 +158,15 @@ public void read(Kryo kryo, Input input) {
         this.toKeep = input.readBoolean();
         boolean hasSource = input.readBoolean();
         if (hasSource) {
-            String clazz = input.readString();
-            Class sourceClass;
             try {
-                sourceClass = Class.forName(clazz);
-                source = (Attribute<?>) sourceClass.newInstance();
-            } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+                String className = input.readString();
+                Class<?> clazz = classCache.get().get(className);
+                if(Attribute.class.isAssignableFrom(clazz)) {
+                    source = (Attribute<?>) clazz.getDeclaredConstructor().newInstance();
+                } else {
+                    throw new RuntimeException("could not instantiate " + className);
```

Comment:
Maybe: `throw new RuntimeException(className " does not extend " + Attribute.getClass().getName());`